### PR TITLE
Update docs for Data Prepper 2.0

### PR DESCRIPTION
**Pull request by oeyh**
### Description
Documentation changes for recent additions in Data Prepper 2.0:
- Core peer forwarder 
- Conditional routing
- A directory for loading pipeline yaml
- CSV processor
- Json processor

### Issues Resolved
N/A

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _clients/data-prepper/data-prepper-reference.md [hunk 1]
```diff
@@ -19,6 +19,51 @@ keyStorePassword | No | String | Password for keystore. Optional, defaults to em
...
+batch_size | No | Integer | Max number of records the buffer returns on read. Default is 48.
+aws_region | Conditionally | String | AWS region to use ACM, S3 or AWS Cloud Map. Required if `use_acm_certificate_for_ssl` is set to true or `ssl_certificate_file` and `ssl_key_file` is AWS S3 path or `discovery_mode` is set to `aws_cloud_map`.
+drain_timeout | No | Duration | Wait time for the peer forwarder to complete processing data before shutdown. Default is 10s.
+
+#### TLS/SSL options for peer forwarder
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+ssl | No | Boolean | Enables TLS/SSL. Default is false.
+ssl_certificate_file | Conditionally | String | SSL certificate chain file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Required if `ssl` is set to true.
```
Comment (dlvenable) [inline _clients/data-prepper/data-prepper-reference.md:52]: > Required if `ssl` is set to true.

I believe we also need `acm_certificate_arn` set to false for this to be required.

File: _clients/data-prepper/data-prepper-reference.md [hunk 2]
```diff
@@ -19,6 +19,51 @@ keyStorePassword | No | String | Password for keystore. Optional, defaults to em
...
+ssl_key_file | Conditionally | String | SSL key file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Required if `ssl` is set to true.
```
Comment (dlvenable) [inline _clients/data-prepper/data-prepper-reference.md:53]: See my comment on `ssl_certificate_file` above regarding being required.

File: _clients/data-prepper/data-prepper-reference.md [hunk 3]
```diff
@@ -379,10 +404,45 @@ Option | Required | Type | Description
...
 with_keys | Yes | List | A list of keys to trim the whitespace from.
 
+### csv
+
+Takes in an Event and parses its CSV data into columns.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+source | No | String | The field in the Event that will be parsed. Default is `message`.
+quote_character | No | String | The character used as a text qualifier for a single column of data. Default is doubel quote `"`.
```
Comment (dlvenable) [inline _clients/data-prepper/data-prepper-reference.md:414]: typo: "doubel" should be "double"

File: _clients/data-prepper/pipelines.md [hunk 4]
```diff
@@ -40,6 +40,38 @@ simple-sample-pipeline:
 
 - Sinks define where your data goes. In this case, the sink is stdout.
 
+Starting from Data Prepper 2.0, pipelines can be defined across multiple configuration YAML files, where each file contains the configuration for one or more pipelines. This gives you more freedom to organize complex pipeline configurations. Those configuration YAML files should be placed in the `pipelines` folder under the application's home directory (e.g. `/usr/share/data-prepper`) for Data Prepper to load properly.
+{: .note }
+
+## Conditional Routing
+
+Pipelines also supports "conditional routing", a feature that allows you to route Events to different sinks based on specific conditions. To add conditional routing to a pipepline, specify a list of named routes under the `route` component and add specific routes to sinks under the `routes` property. Any sink with the routes property will only accept Events which match at least one of the routing condition. For example:
```
Comment (dlvenable) [inline _clients/data-prepper/pipelines.md:48]: I don't think we should quote conditional route.

I'd just say:

> Pipelines also support conditional route, a feature ...

Also, "supports" should be "support."

File: _clients/data-prepper/pipelines.md [hunk 5]
```diff
@@ -234,7 +266,27 @@ serverPort: 1234
...
+docker run --name data-prepper \
+    -v /full/path/to/my-pipelines.yaml:/usr/share/data-prepper/pipelines/my-pipelines.yaml \
+    -v /full/path/to/data-prepper-config.yaml:/usr/share/data-prepper/data-prepper-config.yaml \
     opensearchproject/data-prepper:latest
-````
+```
+
+## Configure the Peer Forwarder
+
+Data Prepper provides an HTTP service to forward `Event` between Data Prepper nodes for aggregation. This is required for operating Data Prepper in a clustered deployment. Currently, peer forwarding is supported in `aggregate`, `service_map_stateful`, and `otel_trace_raw` processors. Peer forwarder groups events based on the identification keys provided by the processors.For `service_map_stateful` and `otel_trace_raw` it's `traceId` by default and can not be configured. For `aggregate` processor, it is configurable using `identification_keys` option.
```
Comment (dlvenable) [inline _clients/data-prepper/pipelines.md:277]: > provided by the processors.For 

Missing space.

File: _clients/data-prepper/pipelines.md [hunk 6]
```diff
@@ -234,7 +266,27 @@ serverPort: 1234
...
+    -v /full/path/to/data-prepper-config.yaml:/usr/share/data-prepper/data-prepper-config.yaml \
     opensearchproject/data-prepper:latest
-````
+```
+
+## Configure the Peer Forwarder
+
+Data Prepper provides an HTTP service to forward `Event` between Data Prepper nodes for aggregation. This is required for operating Data Prepper in a clustered deployment. Currently, peer forwarding is supported in `aggregate`, `service_map_stateful`, and `otel_trace_raw` processors. Peer forwarder groups events based on the identification keys provided by the processors.For `service_map_stateful` and `otel_trace_raw` it's `traceId` by default and can not be configured. For `aggregate` processor, it is configurable using `identification_keys` option. 
+
+Peer forwarder supports peer discovery through one of three options: a static list, a DNS record lookup , or AWS Cloudmap. This option can be configured using `discovery_mode` option. Peer forwarder also supports SSL for verification and encrytion, and mTLS for mutual authentication in peer forwarding service.
```
Comment (dlvenable) [inline _clients/data-prepper/pipelines.md:279]: "AWS Cloud Map"

File: _clients/data-prepper/pipelines.md [hunk 7]
```diff
@@ -234,7 +266,27 @@ serverPort: 1234
...
+
+To configure the peer forwarder, add configuration options to `data-prepper-config.yaml` mentioned in the previous [Configure the Data Prepper server](#configure-the-data-prepper-server) section. Here is an configuration example:
```
Comment (dlvenable) [inline _clients/data-prepper/pipelines.md:281]: > Here is an configuration example:

should be:

> Here is a configuration example:

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: _clients/data-prepper/pipelines.md (hunk 5 above)
Comment (Naarcha-AWS) [inline _clients/data-prepper/pipelines.md:277]:
```suggestion
Data Prepper provides an HTTP service to forward Events between Data Prepper nodes for aggregation. This is required for operating Data Prepper in a clustered deployment. Currently, peer forwarding is supported in `aggregate`, `service_map_stateful`, and `otel_trace_raw` processors. Peer Forwarder groups events based on the identification keys provided by the processors. 

- `service_map_stateful` and `otel_trace_raw` processors use `traceId` by default and therefore can not be configured. 
- The `aggregate` processor is configurable using the `identification_keys` option. 
```

File: _clients/data-prepper/pipelines.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _clients/data-prepper/pipelines.md:281]: Also, no need to introduce the example here, since it comes right after the example's description.

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by oeyh** [comment]
Thank you both for all the feedback and suggestions! Latest commits should resolve most of the comments. I also added some missing options.

File: _clients/data-prepper/data-prepper-reference.md [hunk 8]
```diff
@@ -19,6 +19,54 @@ keyStorePassword | No | String | Password for keystore. Optional, defaults to em
...
+batch_size | No | Integer | Max number of records the buffer returns on read. Default is 48.
+aws_region | Conditionally | String | AWS region to use ACM, S3 or AWS Cloud Map. Required if `use_acm_certificate_for_ssl` is set to true or `ssl_certificate_file` and `ssl_key_file` is AWS S3 path or `discovery_mode` is set to `aws_cloud_map`.
+drain_timeout | No | Duration | Wait time for the peer forwarder to complete processing data before shutdown. Default is 10s.
+
+#### TLS/SSL options for peer forwarder
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+ssl | No | Boolean | Enables TLS/SSL. Default is false.
+ssl_certificate_file | Conditionally | String | SSL certificate chain file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Required if `ssl` is true and `use_acm_certificate_for_ssl` is false.
```
Comment (asifsmohammed) [inline _clients/data-prepper/data-prepper-reference.md:54]: Can you update the `ssl_certificate_file` and `ssl_key_file` with default values? And `ssl` to true by default. These changes are merged in Data Prepper.

Comment (oeyh) [reply _clients/data-prepper/data-prepper-reference.md]: Good catch! Done.

**Review by Naarcha-AWS — APPROVED** [review]

**Review by asifsmohammed — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Pull request merged**